modules/navier_stokes/src/functormaterials/GeneralFunctorFluidProps.C

```diff
@@ -84,14 +90,24 @@ GeneralFunctorFluidProps::GeneralFunctorFluidProps(const InputParameters & param
   // Time derivatives of fluid properties
   //
 
-  addFunctorProperty<ADReal>(NS::time_deriv(NS::density),
-                             [this](const auto & r, const auto & t) -> ADReal
-                             {
-                               ADReal rho, drho_dp, drho_dT;
-                               _fluid.rho_from_p_T(
-                                   _pressure(r, t), _T_fluid(r, t), rho, drho_dp, drho_dT);
-                               return drho_dp * _pressure.dot(r, t) + drho_dT * _T_fluid.dot(r, t);
-                             });
+  addFunctorProperty<ADReal>(
+      NS::time_deriv(NS::density),
+      [this](const auto & r, const auto & t) -> ADReal
+      {
+        if (_neglect_density_time_derivative_jacobian_contributions)
```

Comment:
Can you move the condition outside of the functor material property so that you're choosing between two `addFunctorProperty` calls?

Reply:
done